radicle-cli/src/commands/remote/add.rs

```diff
@@ -0,0 +1,46 @@
+use radicle::node::tracking::store::Config;
+use radicle::{git::Url, node::TRACKING_DB_FILE, prelude::Id, Profile};
+use radicle_crypto::PublicKey;
+
+use crate::{git, terminal as term};
+
+pub fn run(
+    repository: &git::Repository,
+    profile: &Profile,
+    pubkey: &PublicKey,
+    name: &Option<String>,
+    id: Id,
+) -> anyhow::Result<()> {
+    let Some(name) = get_remote_name(profile, pubkey, name)? else {
+        anyhow::bail!("an alias need to be specified");
+    };
+
+    if git::is_remote(repository, &name)? {
+        anyhow::bail!("remote with alias `{name}` already exists");
+    }
+    let url = Url::from(id).with_namespace(*pubkey);
+    let remote = repository.remote(&name, &url.to_string())?;
+    let (name, url) = (
+        remote.name().unwrap_or_default(),
+        remote
+            .url()
+            .ok_or(anyhow::anyhow!("remote url not found"))?,
+    );
```

Comment:
This seems unnecessary since the above `remote` call would have failed otherwise.

If you want to do it for sanity checks then we could do something like this:
```rust
dbg_assert!(remote.name() == Some(&name));
dbg_assert!(remote.url() == Some(url.as_str()));
```

Reply:
@FintanH do we have a consensus about unwrap's and 'SAFETY' comments, or just comments?

Reply:
Oh ya, SAFETY should probably be added, but this would all be rendered moot with this https://github.com/radicle-dev/heartwood/pull/524#discussion_r1171012671

Reply:
Thanks for the suggestion, I think I will add this deb assertion but also I would like to steal some additional information if somethings happens. I added a let else unwrapping to make the branching clean

Regarding the safety comment looks like out of scope, the safety comment needs to be used in unsafe code https://std-dev-guide.rust-lang.org/documentation/safety-comments.html

If it is safe call unwrap on an Option is not something related to `safe rust`, for sure we can add a comment that says "// safe to unwrap because the URL is always Some"

Reply:
> Regarding the safety comment looks like out of scope, the safety comment needs to be used in unsafe code https://std-dev-guide.rust-lang.org/documentation/safety-comments.html
> 
> If it is safe call unwrap on an Option is not something related to `safe rust`, for sure we can add a comment that says "// safe to unwrap because the URL is always Some"

We've documented our own use of SAFETY in [CONTRIBUTING.md](https://github.com/radicle-dev/heartwood/blob/master/CONTRIBUTING.md):

> In the first and second case, document `unwrap` call sites with a comment prefixed
with `SAFETY:` that explains why it's safe to unwrap, eg.
>
>    // SAFETY: Node IDs are valid ref strings.
>    let r = RefString::try_from(node.to_string()).unwrap();

Reply:
> Thanks for the suggestion, I think I will add this deb assertion but also I would like to steal some additional information if somethings happens. I added a let else unwrapping to make the branching clean

That actually defeats the purpose of my suggestion :) I was suggesting using `debug_assert` because that only runs in debug mode, so when we're locally testing -- and so retrieving the URL and name would not be needed when running in production.

Now we're calling those functions twice in debug mode.

So if you're adamant about using those functions then remove the `debug_assert`, but again, IMO they're not needed because the remote should be correct -- it's just `git2` being a pain with the handling of ASCII strings.

Reply:
Oh, the contributing is changes from the last time that I read it! Sorry